### ISSUE TITLE
Absolute layout for inner views doesn't work

### ISSUE THREAD

**Issue by aksonov** [facebook/react-native#2322]
I need to have backdrop background for some modal views - i.e. inner view overlaps parent using 'absolute' position and full-screen dimensions. It worked well with ReactNative 0.6.0, but now doesn't - it just fills parent view. Here is example - instead of full black screen you will see just 200x100 black area.

How to achieve this goal with new version? Maybe i've missed some breaking changes for 0.8.0...

```
       <View style={{flex:1,alignItems: 'center',justifyContent: 'center'}}>
         <Text>Hello world!</Text>
          <View style={{width:200,height:100,backgroundColor:'red'}}>
            <View style={{position:'absolute',top:0,left:0,bottom:0,right:0, backgroundColor:'black'}}/>
          </View>
       </View>
```

**Comment by aksonov** [comment]
"absolute" positioning which is _relative_ to the parent looks quite strange, what is difference between 'relative' and 'absolute' positioning then?

This "improvement" looks quite strange because it brokes encapsulation principles. I can't put modal to somewhere else because button which fires it is inside. Why 'absolute' layout cannot be really absolute?

**Comment by ide** [comment]
It could be, that's just the current behavior of the layout engine: https://github.com/facebook/css-layout

**Comment by aksonov** [comment]
Could you clarify? How to use truly modularized "react" approach with current react-native layout behavior? I want to have "modal" component which could be used from other component. This way i can't put it to 'top-view' as suggested above, it brokes idea of components.

**Comment by browniefed** [comment]
@aksonov there is a `Modal` component coming to core soon, however in the mean time you could use https://github.com/brentvatne/react-native-modal

**Comment by aksonov** [comment]
@browniefed Thanks, I've used it but it doesn't work with 0.8.0 because of issue above.

**Comment by scottgrubhub** [comment]
I am having the same issue. To clarify: from here on out, if you want to position something at 0,0 (top left), the component needs to be directly under the top view? You can't just do {position:absolute, top:0, left:0} ?

**Comment by scottgrubhub** [comment]
To clarify: with newer versions of react, there's no way to position an element at 0,0 (top left of screen), unless that element is a child of the outer most view?

What is the difference between position:relative and position:absolute now?

Thanks

**Comment by browniefed** [comment]
@aksonov `Modal` has now landed as a component in react-native.

@ide @brentvatne closable since Modal is now a thing.

**Comment by ide** [comment]
Also I recommend that you **use a top-level Navigator instead of Modal**. Modal was designed for use in hybrid applications where you have a React Native view embedded in a UINavigationController, and want the React Native code to present a screen on top of the UINavigationController.

With pure React Native, you can simply use a top-level Navigator (wrap your entire app in a scene of this top-level Navigator) and present a modal screen by pushing a scene onto the top-level Navigator's scene stack with the FloatFromBottom config.

**tl;dr: If you are writing a React Native app whose root view is an RCTRootView, do not use Modal. Instead, use Navigator with the "FloatFromBottom" scene config.**

**Comment by ehd** [comment]
@ide Sounds good! Just confirming: Say I already have a Navigator for classic drill-down-and-back history navigation. To get a float-from-bottom modal, I should wrap another top-level navigator around everything, probably one without a NavigationBar, to get the "float from bottom with navigation bar" behaviour?

**Comment by ide** [comment]
@ehd exactly :) just wrap your app in another navigator.
